modin/backends/pandas/query_compiler.py

```diff
@@ -1108,6 +1109,93 @@ def unique(self):
         )
         return self.__constructor__(new_modin_frame)
 
+    def searchsorted(self, **kwargs):
+        """
+        Return a QueryCompiler with value/values indicies, which they should be inserted
+        to maintain order of the passed Series.
+
+        Returns
+        -------
+        PandasQueryCompiler
+        """
+
+        def map_func(part, *args, **kwargs):
+
+            elements_number = len(part.index)
+            assert elements_number > 0, "Wrong mapping behaviour of MapReduce"
+
+            # unify value type
+            value = kwargs.pop("value")
+            value = np.array([value]) if is_scalar(value) else value
+
+            if elements_number == 1:
+                part = part[part.columns[0]]
+            else:
+                part = part.squeeze()
+
+            part_index_start = part.index.start
+            part_index_stop = part.index.stop
+
+            result = part.searchsorted(value=value, *args, **kwargs)
+
+            processed_results = {}
+            value_number = 0
+            for value_result in result:
+                value_result += part_index_start
+
+                if value_result > part_index_start and value_result < part_index_stop:
+                    processed_results[f"value{value_number}"] = {
+                        "allocation": 0,
+                        "index": value_result,
+                    }
```

Comment:
as I understood `allocation` parameter indicates at which partition value should be inserted:
0 - current partition
1 - next partition
-1 - previous partition
can we change parameter name from `allocation` -> `relative_location` or something like this, because `allocation` mostly associates with memory allocation than with location. Also IMHO it would be easier to read if we change 0, 1 and -1 to "current", "next" and "previous"

Reply:
I think it is good suggestion, implemented in the new commits.